src/stash/mod.rs

```diff
@@ -364,20 +364,41 @@ impl<V> Stash<V> {
 
     /// Take an item from a slot (if non empty).
     pub fn take(&mut self, index: usize) -> Option<V> {
-        if let Some(entry) = self.data.get_mut(index) {
-            match mem::replace(entry, Entry::Empty(self.next_free)) {
+        match self.data.get_mut(index) {
```

Comment:
What was your motivation behind this change? I don't really feel strongly one way or another but it doesn't really fit with the PR (unless I'm missing something).